src/components/Offer/Offer.tsx

```diff
@@ -47,6 +49,15 @@ const Offer = ({
     setCoupon(couponCode);
   }, [couponCode]);
 
+  useEffect(() => {
+    // reload Payment component if it was loaded from bfcache
```

Comment:
It would be great if you could add a short note why do we need to do this. Thanks!